test/test_transforms.py

```diff
@@ -30,6 +30,42 @@
     os.path.dirname(os.path.abspath(__file__)), 'assets', 'encode_jpeg', 'grace_hopper_517x606.jpg')
 
 
+def _get_1_channel_tensor_various_types():
+    img_data_float = torch.Tensor(1, 4, 4).uniform_()
+    expected_output = img_data_float.mul(255).int().float().div(255).numpy()
+    yield img_data_float, expected_output, 'L'
+
+    img_data_byte = torch.ByteTensor(1, 4, 4).random_(0, 255)
+    expected_output = img_data_byte.float().div(255.0).numpy()
+    yield img_data_byte, expected_output, 'L'
+
+    img_data_short = torch.ShortTensor(1, 4, 4).random_()
+    expected_output = img_data_short.numpy()
+    yield img_data_short, expected_output, 'I;16'
+
+    img_data_int = torch.IntTensor(1, 4, 4).random_()
+    expected_output = img_data_int.numpy()
+    yield img_data_int, expected_output, 'I'
+
+
+def _get_2d_tensor_various_types():
+    img_data_float = torch.Tensor(4, 4).uniform_()
+    expected_output = img_data_float.mul(255).int().float().div(255).numpy()
+    yield img_data_float, expected_output, 'L'
+
+    img_data_byte = torch.ByteTensor(4, 4).random_(0, 255)
+    expected_output = img_data_byte.float().div(255.0).numpy()
+    yield img_data_byte, expected_output, 'L'
+
+    img_data_short = torch.ShortTensor(4, 4).random_()
+    expected_output = img_data_short.numpy()
+    yield img_data_short, expected_output, 'I;16'
+
+    img_data_int = torch.IntTensor(4, 4).random_()
+    expected_output = img_data_int.numpy()
+    yield img_data_int, expected_output, 'I'
```

Comment:
These two methods have been moved to the top since they are internal use only. They are used in `test_1_channel_tensor_to_pil_image` and `test_2d_tensor_to_pil_image`.

Would it better to add them at the start of class `TestToPILImage`? (probably yes!)
Let me know how that will work.
Say the internal methods are moved into the `TestToPILImage` class, calling these `self._get_xxx_tensor_various_types()` with pytest throws `NameError: name 'self' is not defined` with the following implementation.

```python
class TestToPILImage:

    def _get_1_channel_tensor_various_types(self):
        ....

    @pytest.mark.parametrize('with_mode', [False, True])
    @pytest.mark.parametrize('img_data, expected_output, expected_mode', self._get_1_channel_tensor_various_types())
    def test_1_channel_tensor_to_pil_image(self, with_mode, img_data, expected_output, expected_mode):
        ....

```

Reply:
Never mind this has been resolved now. Though I still have a doubt about why self cannot be passed into a method which is called inside `pytest.mark.parametrize`?